integration_tests/src/scenes/components.rs

```diff
@@ -0,0 +1,198 @@
+#![cfg(test)]
+//! Testing of serialization and deserialization of diverse scene components.
+
+use bevy::ecs::entity::EntityMap;
+use bevy::prelude::*;
+use bevy::reflect::erased_serde::__private::serde::de::DeserializeSeed;
+use bevy::reflect::erased_serde::__private::serde::Serialize;
+use bevy::scene::serde::{SceneDeserializer, SceneSerializer};
+use bevy::scene::ScenePlugin;
+use bincode::Options;
+use std::io::BufReader;
+
+#[test]
+fn ron_roundtrip_equality() {
+    assert_round_trip_equality(serialize_world_ron, deserialize_world_ron);
+}
+
+#[test]
+fn postcard_roundtrip_equality() {
+    assert_round_trip_equality(serialize_world_postcard, deserialize_world_postcard);
+}
+
+#[test]
+fn bincode_roundtrip_equality() {
+    assert_round_trip_equality(serialize_world_bincode, deserialize_world_bincode);
+}
+
+#[test]
+fn messagepack_roundtrip_equality() {
+    assert_round_trip_equality(serialize_world_messagepack, deserialize_world_messagepack);
+}
+
+/// Convenience function for testing the roundtrip equality of different serialization methods.
+fn assert_round_trip_equality(
+    serialize: fn(DynamicScene, &AppTypeRegistry) -> Vec<u8>,
+    deserialize: fn(SceneDeserializer, &[u8]) -> DynamicScene,
+) {
+    let mut input_app = create_test_app();
+    spawn_test_entities(&mut input_app);
+
+    let type_registry = input_app.world.resource::<AppTypeRegistry>();
+    let scene = DynamicScene::from_world(&input_app.world, type_registry);
+    let serialized = serialize(scene, type_registry);
+
+    // We use a clean app to deserialize into, so nothing of the input app can interfere.
+    let mut output_app = create_test_app();
+    let scene = {
+        let scene_deserializer = SceneDeserializer {
+            type_registry: &output_app.world.resource::<AppTypeRegistry>().read(),
+        };
+        deserialize(scene_deserializer, &serialized)
+    };
+
+    let mut entity_map = EntityMap::default();
+    scene
+        .write_to_world(&mut output_app.world, &mut entity_map)
+        .unwrap_or_else(|error| panic!("Could not add deserialized scene to world: {error}"));
+
+    // TODO: Ideally we'd check whether the input and output world are exactly equal. But the world does not implement Eq.
+    //                         so instead we check the serialized outputs against each other. However, this will miss anything that fails to serialize in the first place.
+
+    let type_registry = input_app.world.resource::<AppTypeRegistry>();
+    let scene = DynamicScene::from_world(&input_app.world, type_registry);
+    let serialized_again = serialize(scene, type_registry);
+
+    assert_eq!(serialized, serialized_again);
```

Comment:
maybe you can also check the size of `entity_map` after the call to `write_to_world`?